apps/testing/router-outlet/src/app/app.component.spec.ts

```diff
@@ -1,31 +1,98 @@
+import { userEvent } from '@testing-library/user-event';
+import { render, screen } from '@testing-library/angular';
+import { AppComponent } from './app.component';
+import { appRoutes } from './app.routes';
 describe('AppComponent', () => {
+  const setup = async () => {
+    await render(AppComponent, {
+      routes: appRoutes,
+    });
+    await userEvent.click(screen.getByRole('link', { name: /borrow a book/i }));
+  };
+
+  const searchInput = async (value: string) => {
+    const input = await screen.getByRole('textbox');
+    const borrowBtn = await screen.getByRole('button', { name: 'Borrow' });
+    await userEvent.type(input, value);
+    await userEvent.click(borrowBtn);
+  };
   describe('Given no search criteria', () => {
     it('Then shows error message and disabled button', async () => {
       //todo
+      await setup();
+      const input = screen.getByRole('textbox');
+      await userEvent.clear(input);
+      expect(
+        await screen.queryByText('Search criteria is required!')
+      ).toBeInTheDocument();
```

Comment:
you don't need to add the expect toBeInTheDocument because the `get`method will already assert that the element you are looking for is inside the document, otherwise it wil throw an error.

Reply:
i see, thanks for , i will use
await screen.queryByText('Search criteria is required!')
instead of  expect(
        await screen.queryByText('Search criteria is required!')
      ).toBeInTheDocument();